src/Microsoft.AspNetCore.Mvc.Razor/Compilation/CompiledViewDescriptor.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Compilation
+{
+    public struct CompiledViewDescriptor
```

Comment:
There's really no reason to make this a struct since they get cached. And if you are going to make a struct, immutable, always.